webapp/channels/src/components/plugin_link_tooltip/index.test.tsx

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+
+import {act, renderWithContext, screen, userEvent, waitFor} from 'tests/react_testing_utils';
+import {RootHtmlPortalId} from 'utils/constants';
+
+import PluginLinkTooltip from '.';
+
+function TestLinkTooltip({href}: {href: string}) {
+    if (href.includes('tooltip')) {
+        return <div>{'This is a link tooltip'}</div>;
+    }
+
+    return null;
+}
+
+describe('PluginLinkTooltip', () => {
+    const baseState = {
+        plugins: {
+            components: {
+                LinkTooltip: [
+                    {
+                        id: 'test',
+                        pluginId: 'example.test',
+                        component: TestLinkTooltip as any,
```

Comment:
For some reason, having this be a function component conflicts with `DeepPartial<GlobalState>`. I tried fixing that, but it ended up breaking other tests. I want to cherry-pick this fix to some other branches, so I'll split those off into a separate PR

Reply:
maybe this do not make sense for your test, but what about mocking it using the jest mock function?

Reply:
That might be an option if we don't care what's rendered, but I think that it's more realistic for the test to confirm that the component renders something to the DOM than to just test that the mock function was called for something like this.

Also, I meant to mention it above, but you could also make this work if `TestLinkTooltip` was a class component

Reply:
In retrospect, I'm just going to use a class component for this and call it a day